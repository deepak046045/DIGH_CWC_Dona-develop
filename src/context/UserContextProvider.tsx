'use client';

import React, {
  Dispatch,
  ReactNode,
  SetStateAction,
  createContext,
  useMemo,
  useState,
} from 'react';
import {
  NavigationOption,
  ungatedNavigationOptions,
} from '@/models/NavigationOptions';
import { Appointment } from '@/models/Appointment';
import { AlertType } from '@/models/AlertType';
import mockAppointments from '@/mocks/mockAppointments';
import { CarePartner } from '@/models/CarePartner';
import { mockCarePartners } from '@/mocks/mockCarePartners';

type UserContextDataType = {
  isLoggedIn: boolean;
  setIsLoggedIn: Dispatch<SetStateAction<boolean>>;
  activeView: NavigationOption;
  setActiveView: Dispatch<NavigationOption>;
  alert: AlertType;
  setAlert: Dispatch<SetStateAction<AlertType>>;
  appointments: Appointment[];
  setAppointments: Dispatch<SetStateAction<Appointment[]>>;
  carePartners: CarePartner[];
  setCarePartners: Dispatch<SetStateAction<CarePartner[]>>;
};

export const UserContext = createContext<UserContextDataType>({
  isLoggedIn: false,
  setIsLoggedIn: () => {},
  activeView: ungatedNavigationOptions[0],
  setActiveView: () => {},
  alert: { isOpen: false, message: '', level: '' },
  setAlert: () => {},
  appointments: mockAppointments,
  setAppointments: () => {},
  carePartners: mockCarePartners,
  setCarePartners: () => {},
});

type Props = {
  children: ReactNode;
};

export function UserContextProvider({ children }: Props) {
  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
  const [activeView, setActiveView] = useState<NavigationOption>(
    ungatedNavigationOptions[0]
  );
  const [alert, setAlert] = useState<AlertType>({
    isOpen: false,
    message: '',
    level: '',
  });
  const [appointments, setAppointments] =
    useState<Appointment[]>(mockAppointments);
  const [carePartners, setCarePartners] =
    useState<CarePartner[]>(mockCarePartners);

  const values = useMemo(
    () => ({
      isLoggedIn,
      setIsLoggedIn,
      activeView,
      setActiveView,
      alert,
      setAlert,
      appointments,
      setAppointments,
      carePartners,
      setCarePartners,
    }),
    [isLoggedIn, activeView, alert, appointments, carePartners]
  );

  return <UserContext.Provider value={values}>{children}</UserContext.Provider>;
}
