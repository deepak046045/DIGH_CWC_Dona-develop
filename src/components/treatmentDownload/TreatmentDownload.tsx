'use client';

import React, { useContext } from 'react';
import Image from 'next/image';
import { UserContext } from '@/context/UserContextProvider';
import { LdsButton, LdsHorizontalRule } from '@elilillyco/ux-lds-react';
import { copyTexts } from '@/constants/copyTexts';
import layout from '@/scss/themes/layout.module.scss';
import treatmentDownload from './treatmentDownload.module.scss';

export default function TreatmentDownload() {
  const { isLoggedIn } = useContext(UserContext);
  return (
    <div className={`${treatmentDownload.TreatmentDownload} ${layout.gutter}`}>
      {!isLoggedIn ? (
        <p className={treatmentDownload.TreatmentDownload__text}>
          {copyTexts.TreatmentDownload.reminderUngated}
        </p>
      ) : null}
      <p className={treatmentDownload.TreatmentDownload__text}>
        {isLoggedIn
          ? copyTexts.TreatmentDownload.downloadGated
          : copyTexts.TreatmentDownload.downloadUngated}
        {isLoggedIn ? (
          <LdsButton
            data-testid="print-button"
            classes={`${treatmentDownload['TreatmentDownload__map-download-btn']}`}
          >
            {copyTexts.TreatmentDownload.downloadButtonGated}
            <Image
              src="/icons/printer-icon.svg"
              alt="Printer Icon Download Button"
              width="24"
              height="24"
              className={`${treatmentDownload['TreatmentDownload__map-download-btn-icon']}`}
            />
          </LdsButton>
        ) : (
          <LdsButton
            data-testid="download-button"
            classes={`${treatmentDownload['TreatmentDownload__map-download-btn']}`}
          >
            {copyTexts.TreatmentDownload.downloadButtonUngated}
            <Image
              src="/icons/download-full-icon.svg"
              alt="Download Icon"
              width="19"
              height="19"
              className={`${treatmentDownload['TreatmentDownload__map-download-btn-icon']}`}
            />
          </LdsButton>
        )}
      </p>
      <div className={treatmentDownload.TreatmentDownload__divider}>
        <LdsHorizontalRule />
      </div>
    </div>
  );
}
