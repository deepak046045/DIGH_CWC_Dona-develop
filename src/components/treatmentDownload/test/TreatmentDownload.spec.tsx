import React from 'react';
import { UserContext } from '@/context/UserContextProvider';
import { render, screen } from '@testing-library/react';
import TreatmentDownload from '../TreatmentDownload';

describe('TreatmentDownload', () => {
  it('should render page when logged in', () => {
    const mockValue = {
      isLoggedIn: true,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <TreatmentDownload />
      </UserContext.Provider>
    );

    expect(screen.queryByTestId('print-button')).toBeInTheDocument();
    expect(screen.queryByTestId('download-button')).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the page when not logged in', () => {
    const mockValue = {
      isLoggedIn: false,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <TreatmentDownload />
      </UserContext.Provider>
    );
    expect(screen.queryByTestId('print-button')).not.toBeInTheDocument();
    expect(screen.queryByTestId('download-button')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
