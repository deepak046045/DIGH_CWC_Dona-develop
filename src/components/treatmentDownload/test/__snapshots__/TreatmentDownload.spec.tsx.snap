// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreatmentDownload should render page when logged in 1`] = `
<div>
  <div
    class="TreatmentDownload gutter"
  >
    <p
      class="TreatmentDownload__text"
    >
      Print or save your Treatment Plan as a PDF
      <button
        class="lds-button TreatmentDownload__map-download-btn"
        data-testid="print-button"
        type="button"
      >
        Print My Plan
        <img
          alt="Printer Icon Download Button"
          class="TreatmentDownload__map-download-btn-icon"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/icons/printer-icon.svg"
          style="color: transparent;"
          width="24"
        />
      </button>
    </p>
    <div
      class="TreatmentDownload__divider"
    >
      <lds-hr />
    </div>
  </div>
</div>
`;

exports[`TreatmentDownload should render the page when not logged in 1`] = `
<div>
  <div
    class="TreatmentDownload gutter"
  >
    <p
      class="TreatmentDownload__text"
    >
      Always follow your doctorâ€™s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.
    </p>
    <p
      class="TreatmentDownload__text"
    >
      For a printable version of the treatment map download the following.
      <button
        class="lds-button TreatmentDownload__map-download-btn"
        data-testid="download-button"
        type="button"
      >
        Treatment Map PDF
        <img
          alt="Download Icon"
          class="TreatmentDownload__map-download-btn-icon"
          data-nimg="1"
          decoding="async"
          height="19"
          loading="lazy"
          src="/icons/download-full-icon.svg"
          style="color: transparent;"
          width="19"
        />
      </button>
    </p>
    <div
      class="TreatmentDownload__divider"
    >
      <lds-hr />
    </div>
  </div>
</div>
`;
